### ISSUE TITLE
[Deepspeed alternative] PatrickStar

### ISSUE THREAD

**Issue by flozi00** [huggingface/transformers#17392]
### Feature request

https://github.com/Tencent/PatrickStar

Adding PatrickStar as alternative to deepspeed

### Motivation

I think it could be interesting to benchmark with deepspeed.
In their Readme they are writing it's faster then deepspeed zero 3. 
But they are also writing, gradient accumulation is not possible with the library.

pinging @stas00 for interest ?

### Your contribution

I could give it a try integrating it into the trainer

**Comment by stas00** [comment]
To update - after you shared this a few weeks ago I've requested with the deepspeed devs to look into PatrickStar and see if they could Match the performance - that way we don't need to add a complicated support for another framework. Let's see what unfolds.

**Comment by feifeibear** [comment]
@flozi00 @stas00 Thanks for your attention on PatrickStar. 
I would like to clarify that the comparison results of PatrickStar and DeepSpeed is repeatly varified by many users.
The results in the paper are still valid in May 2022 with the latest version DeepSpeed at that time. 
Also, the PatrickStar idea has already been integrated into ColossalAI and played a key role.
I strongly believe that PatrickStar's design, or part of it, will inherently accelerate the training of large models. I'm open to any help with using Patrickstar, either integrated into DeepSpeed or huggingface transformers.

**Comment by stas00** [comment]
Thank you for your commentary and willingness to contribute, @feifeibear 

> I strongly believe that PatrickStar's design, or part of it, will inherently accelerate the training of large models. I'm open to any help with using Patrickstar, either integrated into DeepSpeed or huggingface transformers.

I think it'd be amazing to have it integrated into Deepspeed. I'm tagging @tjruwase (Deepspeed) on this suggestion

> Also, the PatrickStar idea has already been integrated into ColossalAI and played a key role.

Indeed, we are discussing the CAI integration here: https://github.com/huggingface/transformers/issues/18624

**Comment by feifeibear** [comment]
Thanks! I will keep an eye on the CAI issue! And feel free to contact me if I can help!
